onmt/modules/embeddings.py

```diff
@@ -23,16 +23,29 @@ class PositionalEncoding(nn.Module):
        dim (int): embedding size
     """
 
-    def __init__(self, dim, max_len=5000):
+    def __init__(self, dim, enc_type, max_len=5000):
```

Comment:
Could we make it a kwarg to prevent breaking change on the API?

Reply:
I don't think this specific module is part of any exposed API. The "Embeddings" module itself is, and we have no choice to add the position_encoding_type in Embeddings anyway. PositionalEncoding is exclusively called from Embeddings and IMO it will have to be modified in the future if we want to support learned position encoding or different ones (Rotary, Alibi, ....)